From 8fd579d06da938de4b7d3375dc306d25692285b8 Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Wed, 5 Jun 2024 14:35:35 +0200
Subject: [PATCH 0903/1695] core: scmi-server: fix optee-stm32mp* modules
 directives

Remove conditional use of scpfw-embed-product-module macro in
optee-stm32mp1 and optee-stm32mp2 SCP-firmware products to
ensure the directive CFG_SCPFW_MOD_* are provided to SCP-firmware
CMake configuration unless what SCP-firmware generate files
fwk_module_list.c and fwk_module_idx.h are not generated with update
config switches for these product specific modules.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git c4cd8e8e26683087257db0a4ccf49c2e4f0719f2 core: scmi-server: fix optee-stm32mp* modules directives]
Change-Id: If291eecbf191042704037c394327b3114a36db58
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/385417
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 core/lib/scmi-server/sub-optee-stm32mp1.mk | 2 --
 core/lib/scmi-server/sub-optee-stm32mp2.mk | 6 ------
 2 files changed, 8 deletions(-)

diff --git a/core/lib/scmi-server/sub-optee-stm32mp1.mk b/core/lib/scmi-server/sub-optee-stm32mp1.mk
index 8555faae4..223810280 100644
--- a/core/lib/scmi-server/sub-optee-stm32mp1.mk
+++ b/core/lib/scmi-server/sub-optee-stm32mp1.mk
@@ -2,8 +2,6 @@ scpfw-incdirs-y += $(scpfw-path)/product/optee-stm32mp1/include
 
 srcs-y += $(scpfw-path)/product/optee-stm32mp1/fw/config_all.c
 
-ifeq ($(CFG_SCPFW_MOD_PSU_OPTEE_REGULATOR),y)
 $(eval $(call scpfw-embed-product-module,psu_optee_regulator))
-endif
 
 scpfw-cmake-flags-y += -DCFG_STM32MP13=$(CFG_STM32MP13)
diff --git a/core/lib/scmi-server/sub-optee-stm32mp2.mk b/core/lib/scmi-server/sub-optee-stm32mp2.mk
index bba6e7671..d3fcc9d0e 100644
--- a/core/lib/scmi-server/sub-optee-stm32mp2.mk
+++ b/core/lib/scmi-server/sub-optee-stm32mp2.mk
@@ -2,11 +2,5 @@ scpfw-incdirs-y += $(scpfw-path)/product/optee-stm32mp2/include
 
 srcs-y += $(scpfw-path)/product/optee-stm32mp2/fw/config_all.c
 
-ifeq ($(CFG_SCPFW_MOD_PSU_OPTEE_REGULATOR),y)
 $(eval $(call scpfw-embed-product-module,psu_optee_regulator))
-endif
-
-ifeq ($(CFG_SCPFW_MOD_STM32_PD),y)
 $(eval $(call scpfw-embed-product-module,stm32_pd))
-endif
-
-- 
2.47.2

